core/src/main/java/org/jdbi/v3/ResultBearing.java

```diff
@@ -63,4 +63,15 @@ default ResultType findOnly() {
             return stream.collect(Collectors.<ResultType>toList());
         }
     }
+
+    /**
+     * Collect the query result into a container of the type specified by the given class.
+     * A factory for building the container should be registered in the query's
+     * container factory registry.
+     *
+     * @param containerType   the class that represents the container
+     * @param <ContainerType> the generic type of the container
+     * @return the container with the query result
+     */
+    <ContainerType> ContainerType collect(Class<ContainerType> containerType);
```

Comment:
Let's rename this to `collectInto(Class<ContainerType>)` and an additional `collect(Collector<T,A,R>)` method alongside it.